### ISSUE TITLE
[Bug] showLastSeenWhenOffline

### ISSUE THREAD

**Issue by TimerWolf** [doctorfree/MMM-MacAddressScan#7]
I do not think this option works as it should, i have not look in it a lot but if i change the the code from

```js
if ((self.config.showLastSeen && device.lastSeen  && !self.config.showLastSeenWhenOffline) || 
   (self.config.showLastSeen && !device.lastSeen &&  self.config.showLastSeenWhenOffline)) {
```

to the fallowing code

```js
if (self.config.showLastSeen && !device.online) {
```

Then it works as "showLastSeenWhenOffline" should have been working that is "show last seen only when offline." if you look at the description.

**Comment by doctorfree** [comment]
Good catch. My interpretation is a little different but you are right about the "show last seen only when offline" case.

There are two configuration options, `showLastSeen` and `showLastSeenWhenOffline`, both booleans. That means there are four configuration states, both true, both false, and one true but the other false. What do we want to see for each configuration state? If both are false then it's easy, don't show last seen for any device. If both are true then I think we want what you suggest. But how to handle the other two cases? I think we have to cover the case where `showLastSeen` is true but `showLastSeenWhenOffline` is false which would show last seen for only online devices. The final case would not show any last seen info, similar to both false.

Does that make sense to you?

Thanks for opening this issue, the logic in that section of code is faulty.

**Comment by doctorfree** [comment]
Actually, maybe what I would like is to be able to see last seen for all devices sometimes. Crap.

**Comment by doctorfree** [comment]
@TimerWolf i am just going to make these do what they are documented to do and not worry about other states.

That is, if show last seen is enabled then there are two states determined by `showLastSeenWhenOffline`, either show all devices' last seen status or only show last seen when offline (as you have noted).

I'll get to this as soon as I can.
